### ISSUE TITLE
Suggestion: add developer friendly resources

### ISSUE THREAD

**Issue by Decagon** [Tunous/EverybodyEditsProtocol#3]
Adding in developer friendly resources will help keep EE-related bots and applications up to date at a click of the button, without having to update the actual application, just the related files. Developers can just add the URL of the file to their application from a trusted repository, and update it client-side when there's a new EE update.

For example, [this](https://github.com/Decagon/EEBlocks/blob/master/Other/payvault.yaml) file is able to be parsed by C# using the YAMLDotNet library, and perfectly parses it into an array, which is able to be iterated over. YAML is much more readable than XML, and is able to be parsed 100% without errors. 

Other formats, like using the equals sign, makes the user have to create their own parser, and there are lots of edge cases that are not planned for (for example, adding a comment to a line) which cause applications to crash.

**Comment by Tunous** [comment]
I agree about this. I'm definitely willing to accept any pull request that add any useful resources related to bot programming. 

Perhaps current shop data and models from the main readme could be moved into seaparate files with easy to parse format. For that I would like to hear opinions angle what format is our best choice. I personally also like yaml so that would be my vote. 

If we decided on one format and added some of resources short instructions about how to parse them would be also a great addition. 

So that gives us 5 main questions. 
1. What format to use? 
2. What resources to add? 
3. Would we prefer to store them in few bigger files or in one file for each specific resource? 
4. Where we should put parsing instructions?

**Comment by Decagon** [comment]
We should add in the following for resources:

- block images (used in the wiki, [ee-colorwheel](http://ee-colorwheel.herokuapp.com/), ee-apps like EEPainter, the forums, and occasionally the blog)
- block colors (minimap colors)
- sound blocks (used in midi apps)
- smiley images
- payvault shop data (can be programatically generated, used in the wiki, e.g [here](https://github.com/Decagon/EEBlocks/blob/master/Other/payvault.yaml))
- links (or tutorials) to create an ee bot, and links to various sdk's

As for storing the data, storing it in one file might be best (as programmatically downloading one file is easier to discover newer features than discovering new files.)

Parsing instructions should be in a seperate file, linked to in the README.md file.

**Comment by Decagon** [comment]
Which resources did you want to add? I can send a pull request when I get them prepared.

**Comment by Tunous** [comment]
I think that minimap colors, payvault shop data and perhaps links to tutorials are enough. Images aren't that useful here as it's more about the code than graphics. Plus there is already https://github.com/EEJesse/EEBlocks

**Comment by Tunous** [comment]
Blocks, smileys and auras data is almost ready in Unity EE. I'll be using it to automatically generate publicly accessible version.

As I see only minimap colors are still incorrect for some of the blocks. I'll work on creating a code to generate it and will start a pull request when initial version is ready, then we can take few weeks to fix any issues and merge it once finished.

The data will be presented in YAML format.

**Comment by SirJosh3917** [comment]
I've tried developing a readme for shop colors, you can look at it, un-complete, from [here.](https://github.com/SirJosh3917/EverybodyEditsProtocol)

**Comment by Tunous** [comment]
@SirJosh3917 What do you mean by "shop colors"?

**Comment by Tunous** [comment]
About my previous comment: if someone wonders why I still didn't add that Unity generated data it's because I decided to not use it before the release of Unity EE.
